src/lib/containers/DiscussionSearchResult.tsx

```diff
@@ -35,51 +36,69 @@ const DiscussionSearchResult = (props: DiscussionSearchResultProps) => {
     const [messageUrl, setMessageUrl] = useState<string>('')
     const [replyBundle, setReplyBundle] = useState<DiscussionReplyBundle>()
     const [replyAuthor, setReplyAuthor] = useState<UserProfile>()
+    const [isLoading, setIsLoading] = useState(false)
 
     const getThreadOrReply = async () => {
-        let someUrl
+        let newMessageUrl
         const thread = await getThread(threadId, accessToken)
+        setIsLoading(true)
         if (replyId) {
             const reply = await getReply(replyId, accessToken)
-            someUrl = await getReplyMessageUrl(reply.messageKey, accessToken)
+            newMessageUrl = await getReplyMessageUrl(reply.messageKey, accessToken)
             setReplyAuthor(await getUserProfileById(accessToken, reply.createdBy))
             setReplyBundle(reply)
         } else {
             setReplyAuthor(await getUserProfileById(accessToken, thread.createdBy))
-            someUrl = await getThreadMessageUrl(thread.messageKey, accessToken)
+            newMessageUrl = await getThreadMessageUrl(thread.messageKey, accessToken)
         }
-        setMessageUrl(await getMessage(someUrl.messageUrl))
+        setMessageUrl(await getMessage(newMessageUrl.messageUrl))
         setThreadBundle(thread)
+        setIsLoading(false)
     }
 
     useEffect(() => {
         getThreadOrReply()
     }, [])
 
-    const getUrl = (threadId: string, replyId: string, projectId: string) => {
+    const getUrl = (threadId: string, projectId: string, replyId?: string,) => {
         return `https://synapse.org/#!Synapse:${projectId}/discussion/threadId=${threadId}&replyId=${replyId}`
```

Comment:
Please only include the query param replyId if replyId is set (do not include replyId=undefined)